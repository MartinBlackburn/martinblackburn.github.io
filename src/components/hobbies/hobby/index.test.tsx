// libraries
import { render } from "@testing-library/react";

// component under test
import Hobby from "./";

describe("Hobby component", () => {
    test("renders correctly", () => {
        const TextComponent = () => {
            return <p>some text about climbing</p>;
        };

        const { asFragment } = render(<Hobby title="Climbing" image="climbing.png" text={<TextComponent />} />);

        expect(asFragment()).toMatchSnapshot();
    });
});
