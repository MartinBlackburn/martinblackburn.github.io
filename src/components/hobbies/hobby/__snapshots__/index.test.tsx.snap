// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hobby component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="hobby"
  >
    <h2
      class="hobby__title"
    >
      Climbing
    </h2>
    <div
      class="hobby__image"
    >
      <img
        alt="illustration of someone Climbing"
        src="/images/hobbies/climbing.png"
      />
    </div>
    <div
      class="hobby__content"
    >
      <p>
        some text about climbing
      </p>
    </div>
  </div>
</DocumentFragment>
`;
